operator/webhooks/pod_mutating_webhook.go

```diff
@@ -72,3 +85,126 @@ func (a *PodMutatingWebhook) isKeptnAnnotated(pod *corev1.Pod) bool {
 	}
 	return false
 }
+
+func (a *PodMutatingWebhook) calculateVersion(pod *corev1.Pod) string {
+	name := ""
+	for _, item := range pod.Spec.Containers {
+		name = name + item.Name + item.Image
+		for _, e := range item.Env {
+			name = name + e.Name + e.Value
+		}
+	}
+
+	h := fnv.New32a()
+	h.Write([]byte(name))
+	return fmt.Sprint(h.Sum32())
+}
+
+func (r *PodMutatingWebhook) handleServiceRun(ctx context.Context, logger logr.Logger, pod *corev1.Pod, namespace string) error {
+	serviceName, _ := pod.Annotations["keptn.sh/service"]
+
+	logger.Info("Service name", "service", serviceName)
+
+	service := &v1alpha1.Service{}
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: namespace, Name: serviceName}, service)
+	if errors.IsNotFound(err) {
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("could not fetch Service: %+v", err)
+	}
+
+	serviceRun := &v1alpha1.ServiceRun{}
+	err = r.Client.Get(ctx, types.NamespacedName{Namespace: service.Namespace, Name: service.GetServiceRunName()}, serviceRun)
+	if errors.IsNotFound(err) {
+		logger.Info("Creating serviceRun from service", "service", service.Name)
+		serviceRun, err := r.createServiceRun(ctx, service)
+		if err != nil {
+			logger.Error(err, "Could not create ServiceRun")
+			return err
+		}
+
+		k8sEvent := r.generateK8sEvent(service, serviceRun)
+		if err := r.Client.Create(ctx, k8sEvent); err != nil {
+			logger.Error(err, "Could not send serviceRun created K8s event")
+			return err
+		}
+
+		if err := r.Client.Status().Update(ctx, service); err != nil {
+			logger.Error(err, "Could not update Service")
+			return err
+		}
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("could not fetch ServiceRun: %+v", err)
+	}
+
+	return nil
+}
+
+func (r *PodMutatingWebhook) createServiceRun(ctx context.Context, service *v1alpha1.Service) (*v1alpha1.ServiceRun, error) {
```

Comment:
The webhook should create a service (not a servicerun)

Reply:
from where should the webhook know what kind of pre-checks (in our case jobs) should be executed? This information in now store in the service, under the spec section, also, the serviceRun should be bind to a specific version of the pod/deployment/replicaset... There should be a 1on1 binding between the version of a pod and a service run (real execution of the tests)

Reply:
Or am I missing some information? Can we maybe discuss this?